### ISSUE TITLE
`new super()` in static method does not cause error

### ISSUE THREAD

**Issue by Withered-Flower-0422** [microsoft/typescript-go#4704]
This is from [TS6.0](https://github.com/microsoft/TypeScript/issues/63451).
Maybe can be fixed in tsgo?

### 💻 Code

```ts
class A extends class {} {
    static {
        new super()
    }

    static f() {
        new super()
    }
}
```

### 🙁 Actual behavior

no error

### 🙂 Expected behavior

`new super()` is invalid and should throw error.

**Comment by RyanCavanaugh** [comment]
We can continue tracking this at the linked issue. Thanks!
